index.bs

```diff
@@ -560,79 +560,251 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p>
+			The {{ClipboardItem/types}} getter steps are to return [=this=]'s [=ClipboardItem/types array=].
+		</p>
+
+		<h4 method for=ClipboardItem lt=getType(type)>getType(type)</h4>
+		
+			This method must run the below steps:
+
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |mimeType| be the result of [=parsing a MIME type=] given |type|.
+
+			1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+			1. Let |itemTypeList| be [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+			1. Let |p| be [=a new promise=] in |realm|.
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+			1. For each |representation| in |itemTypeList|:
+
+				1. If |representation|'s [=representation/MIME type=] is |mimeType|, then:
+
+					1. Let |representationDataPromise| be the |representation|'s [=representation/data=].
+
+					1. [=promise/React=] to |representationDataPromise|:
+
+						1. If |representationDataPromise| was fulfilled with value |v|, then:
+
+							1. If |v| is a {{DOMString}}, then follow the below steps:
+
+								1. Let |dataAsBytes| be the result of [=UTF-8 encoding=] |v|.
+
+								1. Let |blobData| be a {{Blob}} created using |dataAsBytes| with its {{Blob/type}} set to |mimeType|, [=serialize a MIME type|serialized=].
+
+								1. Resolve |p| with |blobData|.
+
+							1. If |v| is a {{Blob}}, then follow the below steps:
+
+								1. Resolve |p| with |v|.
+
+						1. If |representationDataPromise| was rejected, then:
+
+							1. [=Reject=] |p| with {{"NotFoundError"}} {{DOMException}} in |realm|.
+
+					1. Return |p|.
+
+			1. [=Reject=] |p| with {{"NotFoundError"}} {{DOMException}} in |realm|.
+
+			1. Return |p|.
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		The methods of the {{Clipboard}} interface take or return multiple {{ClipboardItem}} objects, and their specification is written to deal with the generic case. However, not all platforms support more than one [=/clipboard item=]; on such platforms, the algorithms below will ignore any {{ClipboardItem}} objects beyond the first one that are passed to {{Clipboard/write()}}, and {{Clipboard/read()}} and {{Clipboard/readText()}} will only ever return a single-item array.
+
+		A {{Clipboard}} object has an associated <dfn>clipboard</dfn>, which is a [=clipboard=].
```

Comment:
This doesn't make sense the last "clipboard" again links to the `Clipboard` defined in the IDL. As that `clipboard` seems never used, presumably this sentence can be deleted.

Reply:
This is the conceptual `clipboard` object. Deleted the part that links to the IDL.